lib/chef/knife/bootstrap_windows_base.rb

```diff
@@ -319,6 +322,23 @@ def bootstrap(proto=nil)
         end
 
         wait_for_remote_response( config[:auth_timeout].to_i )
+
+        # We allow the user to specify the desired architecture of Chef to install or we default
+        # to whatever the target system is
+        session_results = relay_winrm_command("echo %PROCESSOR_ARCHITECTURE%")
+        current_architecture = session_results[0][:data][0][:stdout] == "X86" ? :i386 : :x86_64
```

Comment:
We should examine each potential node separately instead of basing the architecture for all nodes from the results of the first.